### ISSUE TITLE
Evaluate better crates

### ISSUE THREAD

**Issue by zhangsoledad** [cryptape/cita-vm#13]
Evaluate [numext](https://github.com/yangby-cryptape/rust-numext) as a replacement for `ethereum-types`,
Introduce [faster-hex](https://github.com/nervosnetwork/faster-hex) to do the hex encoding through SIMD intrinsics.

**Comment by mohanson** [comment]
`hex` is only used for tests, so it's unnecessary to evaluate. I will evaluate if `numext` can be integrated.

**Comment by u2** [comment]
Need `numext` supporting conversion from `fixed_uint` to u64.

**Comment by ouwenkg** [comment]
A sadly thing, I encountered all kind of problems when replacing [ethereum-types](https://crates.io/crates/ethereum-types) with [numexe-fix-uint](https://crates.io/crates/numext-fixed-uint) and [numext-fixed-hash](https://crates.io/crates/numext-fixed-hash) in my branch [Replace_ethereum_types](https://github.com/ouwenkg/cita-vm/tree/replace_ethereum_types)

Tricks List:
* Rlp encode/decode, [Rlp](https://crates.io/crates/rlp) automatically impements `encode` and `decode` trait for `ethereum_types` Using `numext-*`, we must finish this work handly, well, i solve it in my branch.

* `H160 -> H256` or `H256 -> H160`, `numext-*` relevant transfer functions behave differ from ethereum_types. CITA-VM transfer funtions needed. solved well.

* Big endian or Little endian, this trick happened everywhere.

In summary, this update may not be necessary, a tough modification may bring some unexpected situations, so i will not continue to do this task.

**Comment by mohanson** [comment]
Ok. Thanks for your work.
